src/components/MaterialUI/DataDisplay/Table.js

```diff
@@ -1,64 +1,361 @@
+			/>
+		</td>
+	);
+};
+
+const buildTableCheckbox = (classes, tableSelectionStatus, selectionMethods) => {
+	return (
+		<th key="tableSelector" className={classes.headerCellSelect}>
+			<CheckboxMui
+				className={classes.rowSelectCheckbox}
+				checked={tableSelectionStatus === tableSelectionMode.all}
+				indeterminate={tableSelectionStatus === tableSelectionMode.indeterminate}
+				onChange={event => selectionMethods.selectionHandler(event, null)}
+			/>
+		</th>
+	);
+};
+
+const StickerTableHeader = React.forwardRef((props, refHeader) => (
+	<div key="stickyHeader" className={props.classes.stickyHeaderHead}>
+		<TableMui className={props.classes.stickyHeaderTable}>
+			<TableHead ref={refHeader}>
+				<TableRow>{props.tableHeaders}</TableRow>
+			</TableHead>
+		</TableMui>
+		<div className={props.classes.stickyHeaderTableScroll} />
+	</div>
+));
+
+const buildTableHeaders = (
+	headers,
+	classes,
+	selectMode,
+	tableSelectionStatus,
+	selectionMethods,
+) => {
+	const tableCheckbox =
+		selectMode === true
+			? [buildTableCheckbox(classes, tableSelectionStatus, selectionMethods)]
+			: [];
+
+	return tableCheckbox.concat(
+		headers.map((header, index) => (
+			<th key={index} className={classes.headerCell} style={header.cellStyle}>
+				{header.cellElement}
+			</th>
+		)),
+	);
+};
+
+const buildTableRows = (rows, classes, selectMode, onRowClick, selectionHandlers) => {
+	const onClick = (evt, row) => {
+		if (evt.target.tagName !== "INPUT") {
+			onRowClick(evt, row.element);
+		}
+	};
+
+	return rows.map(row => (
+		<MemoTableRow
+			className={classes.tableRow}
+			key={row.key}
+			onClick={evt => onClick(evt, row)}
+			selected={selectionHandlers.isSelected(row.key)}
+		>
+			{selectMode === true ? buildRowCheckbox(classes, row.key, selectionHandlers) : null}
+			{row.columns.map((cell, cellIndex) => (
+				<MemoTableCell
+					className={classes.tableCell}
+					key={cellIndex}
+					value={cell.cellElement}
+					titleValue={cell.title}
+					style={cell.cellStyle}
+				/>
+			))}
+		</MemoTableRow>
+	));
+};
+
+const FullTable = React.forwardRef((props, ref) => {
+	const scrollEvent = evt => {
+		if (
+			evt.target.scrollHeight - (evt.target.scrollTop + evt.target.offsetHeight) < 100 &&
+			props.rows.length === props.latestPage * props.pageLength
+		) {
+			props.scrollLoader(props.latestPage + 1);
+		}
+	};
+
+	return (
+		<div
+			key="actualTable"
+			className={props.classes.tableContainer}
+			ref={ref}
+			onScroll={scrollEvent}
+		>
+			<TableMui className={props.classes.table}>
+				<TableHead
+					className={props.classes.tableHeader}
+					ref={props.stickyHeader ? null : ref}
+				>
+					<TableRow>{props.tableHeaders}</TableRow>
+				</TableHead>
+				<MemoTableBody
+					className={props.classes.tableBody}
+					contentTimestamp={props.contentTimestamp}
+					rows={props.rows}
+					selectedNumber={props.selectedNumber}
+				/>
+			</TableMui>
+		</div>
+	);
+});
+
+const Table = ({
+	contentTimestamp,
```

Comment:
is that possible to avoid that parameter? as a user of table I don't really want pass parameters like this. is that possible to get timestamp just inside the table?

Reply:
There is a use case when Filter Bar is opened and it triggers the scroll bar to appear on the table. I would need an resize event listener on the DOM element, which is not supported out-of-the-box by most browsers. What would you think about the package "https://github.com/marcj/css-element-queries" ? Let me know please

Reply:
Well, if in this case we need to change only styles based on resize event then I would say it would be better to have packages like this

Reply:
Let me try it first, if it works, I will add it in orc-script as well as why-did-you-rerender while I am at it.